### PR TITLE
add EventEmitter example

### PR DESCRIPTION
**Pull request by nicolaspanel**
As requested in #5 (and more general than https://github.com/bnoordhuis/node-event-emitter/)

### CODE REVIEW

**Comment by eljefedelrodeodeljefe** [comment]
hmm. Looks a little verbose. The goal is interesting though. @kkoopa @rvagg do you have opnions on that one? You have participated in the issue. cc @bnoordhuis due to [bnoordhuis/node-event-emitter](https://github.com/bnoordhuis/node-event-emitter)

File: 9_event_emitter/nan/timer.cc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  Nan::SetPrototypeMethod(tpl, "start", Start);
+  Nan::SetPrototypeMethod(tpl, "stop", Stop);
+
+  constructor.Reset(tpl->GetFunction());
+  exports->Set(Nan::New("Timer").ToLocalChecked(), tpl->GetFunction());
+}
+
+void Timer::New(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  if (!info.IsConstructCall()) {
+    Nan::ThrowError("`new` required");
```
Comment (kkoopa) [inline 9_event_emitter/nan/timer.cc]: missing return

File: 9_event_emitter/nan/timer.cc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  obj->emit = new Nan::Callback(
+    v8::Local<v8::Function>::Cast(obj->handle()->Get(Nan::New("emit").ToLocalChecked())));
+  info.GetReturnValue().Set(info.This());
+}
+
+void Timer::Start(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  Timer* obj = Nan::ObjectWrap::Unwrap<Timer>(info.Holder());
+  if (obj->timer_req_ != nullptr) {
+    // already started
+    info.GetReturnValue().Set(Nan::Undefined());
```
Comment (kkoopa) [inline 9_event_emitter/nan/timer.cc]: unnecessary

File: 9_event_emitter/nan/timer.cc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    return;
+  }
+  uint64_t timeout = info[0]->IsUndefined() ? 0 : info[0]->NumberValue();
+  uint64_t repeat = info[1]->IsUndefined() ? 0 : info[1]->NumberValue();
+  uv_timer_t *tr = new uv_timer_t;
+  obj->timer_req_ = tr;
+  tr->data = static_cast<void*>(obj);
+  uv_timer_init(uv_default_loop(), tr);
+  uv_timer_start(tr, onTick, timeout, repeat);
+  info.GetReturnValue().Set(Nan::Undefined());
```
Comment (kkoopa) [inline 9_event_emitter/nan/timer.cc]: unnecessary

File: 9_event_emitter/nan/timer.cc [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  if (obj->timer_req_ != nullptr) {
+    // already started
+    info.GetReturnValue().Set(Nan::Undefined());
+    return;
+  }
+  uint64_t timeout = info[0]->IsUndefined() ? 0 : info[0]->NumberValue();
+  uint64_t repeat = info[1]->IsUndefined() ? 0 : info[1]->NumberValue();
+  uv_timer_t *tr = new uv_timer_t;
+  obj->timer_req_ = tr;
+  tr->data = static_cast<void*>(obj);
```
Comment (kkoopa) [inline 9_event_emitter/nan/timer.cc]: unnecessary cast

**Comment by nicolaspanel** [comment]
@eljefedelrodeodeljefe: thank you for the review

**Comment by dsamarin** [comment]
Why hasn't this merged yet? It's very useful to me

**Comment by mhdawson** [comment]
We don't have reviewers for the nan part anymore and this has been open for 4 years. I'm going to close, please us know if you think this is not the right thing to do.

**Pull request closed without merging**